pandas/core/arrays/datetimelike.py

```diff
@@ -1025,9 +1025,8 @@ def _addsub_object_array(self, other: np.ndarray, op):
         result : same class as self
         """
         assert op in [operator.add, operator.sub]
-        if len(other) == 1:
+        if len(other) == 1 and self.ndim == 1:
```

Comment:
@jbrockmendel do you recall if tests were failing without this change, or unrelated to the tests added here.

Reply:
Yes, a test failed without this change

Reply:
on the backport tests fail with and without this change. with this change, we get an additional 13 performance warnings. without this change, we don't get performance warnings in 3 tests

Reply:
> without this change, we don't get performance warnings in 3 tests

Probably the same three I saw on master that motivated this edit.  I just checked out the backport branch locally, will try to troubleshoot.

Reply:
im not wild about not-having a better solution, but i think the immediate issue can be addressed by making the condiiton `and (self.ndim == 1 or other.shape == self.shape == (1, 1))`

Reply:
Thanks @jbrockmendel for looking into this. I have updated the backport PR with this change for now, pending further discussion there on best approach. https://github.com/pandas-dev/pandas/pull/37633#issuecomment-723511112

Reply:
the suggested update applied to the backport still failed. https://github.com/pandas-dev/pandas/pull/37633/commits/458e0ce754c121fdd7c0792a0f8ed80742e1dafb

Reply:
@jbrockmendel I'm struggling with this. (working on 1.1x so maybe differences with master)

can you justify this change other than tests were failing.

somethings changed in the internals dispatch where other in the _addsub_object_array call is now a scalar (in a 2d array with shape 1,1)  for the failing `test_td64arr_add_sub_object_array` test instead of a 2d array on 1.1.x

so not raising a performance warning makes sense if other is now a 'scalar'

(also the repr fails for a 2d timedelta array)

on 1.1.x
```
(Pdb) other
array([[Timedelta('1 days 00:00:00')],
       [<2 * Days>]], dtype=object)
(Pdb) a
Traceback (most recent call last):
...
ValueError: Value must be Timedelta, string, integer, float, timedelta or convertible, not TimedeltaArray

(Pdb) self._data
array([[ 86400000000000],
       [172800000000000]], dtype='timedelta64[ns]')
(Pdb)
```

with the patch applied ( #37633)
```
c:\users\simon\pandas\pandas\core\arrays\datetimelike.py(1337)_addsub_object_array()
-> assert op in [operator.add, operator.sub]
(Pdb) other
array([[<2 * Days>]], dtype=object)
(Pdb) self._data
array([[172800000000000]], dtype='timedelta64[ns]')
(Pdb)
```

Reply:
> can you justify this change other than tests were failing.

I never did manage to track down the underlying source of the problem that made this particular edit necessary.  Its still early-ish afternoon for me so I can spend some more time today trying again if you think its a priority.

> (also the repr fails for a 2d timedelta array)

FWIW that was fixed in #37164